components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/GraphQLSchemaDefinition.java

```diff
@@ -197,4 +214,90 @@ protected void addQueryParams(JSONObject pathsObject) {
+            resource.setMediaType(String.valueOf(ContentType.TEXT_PLAIN));
+            registry.put(saveResourcePath, resource);
+            if (log.isDebugEnabled()) {
+                log.debug("Successfully imported the schema: " + schemaDefinition );
+            }
+
+            String[] visibleRoles = null;
+            if (api.getVisibleRoles() != null) {
+                visibleRoles = api.getVisibleRoles().split(",");
+            }
+
+            //Need to set anonymous if the visibility is public
+            APIUtil.clearResourcePermissions(resourcePath, api.getId(), ((UserRegistry) registry).getTenantId());
+            APIUtil.setResourcePermissions(apiProviderName, api.getVisibility(), visibleRoles, resourcePath);
+
+        } catch (RegistryException e) {
+            String errorMessage = "Error while adding Graphql Definition for " + apiName + '-' + apiVersion;
+            log.error(errorMessage, e);
+            handleException(errorMessage, e);
+        }
+    }
+
+    /**
+     * Returns the graphQL content in registry specified by the wsdl name
+     *
+     * @param apiId Api Identifier
+     * @return graphQL content matching name if exist else null
+     */
+    public String getGraphqlSchemaDefinition(APIIdentifier apiId, Registry registry) throws APIManagementException {
+        String apiName = apiId.getApiName();
+        String apiVersion = apiId.getVersion();
+        String apiProviderName = apiId.getProviderName();
+        String resourcePath = APIUtil.getGraphqlDefinitionFilePath(apiName, apiVersion, apiProviderName);
+
+        String schemaDoc = null;
+        String schemaName = apiId.getProviderName() + APIConstants.GRAPHQL_SCHEMA_PROVIDER_SEPERATOR +
+                apiId.getApiName() + apiId.getVersion() + APIConstants.GRAPHQL_SCHEMA_FILE_EXTENSION;
+        String schemaResourePath = resourcePath + schemaName;
+        try {
+            if (registry.resourceExists(schemaResourePath)) {
+                Resource schemaResource = registry.get(schemaResourePath);
+                schemaDoc = IOUtils.toString(schemaResource.getContentStream(),
+                        RegistryConstants.DEFAULT_CHARSET_ENCODING);
+            }
+        } catch (org.wso2.carbon.registry.core.exceptions.RegistryException e) {
+            String msg = "Error while getting schema file from the registry " + schemaResourePath;
+            log.error(msg, e);
+            throw new APIManagementException(msg, e);
+        } catch (IOException e) {
+            String error = "Error occurred while getting the content of schema " + schemaName;
+            log.error(error);
+            throw new APIManagementException(error, e);
+        } catch (RegistryException e) {
+            String msg = "Failed to get swagger documentation of API : " + apiId;
+            log.error(msg, e);
+            throw new APIManagementException(msg, e);
+        }
+        return schemaDoc;
+    }
 }
```

Comment:
Add a new line at EOF

Reply:
This function has been implemented end of the file. So, We don't want to add a new line in here